pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -244,3 +432,109 @@ func (r *MachineSyncReconciler) shouldMirrorCAPIMachineToMAPIMachine(ctx context
 
 	return false, nil
 }
+
+// fetchCAPIInfraResources fetches the provider specific infrastructure resources depending on which provider is set.
+func (r *MachineSyncReconciler) fetchCAPIInfraResources(ctx context.Context, capiMachine *capiv1beta1.Machine) (client.Object, client.Object, error) {
+	var infraCluster, infraMachine client.Object
+
+	infraClusterKey := client.ObjectKey{
+		Namespace: capiMachine.Namespace,
+		Name:      capiMachine.Spec.ClusterName,
+	}
+
+	infraMachineRef := capiMachine.Spec.InfrastructureRef
+	infraMachineKey := client.ObjectKey{
+		Namespace: infraMachineRef.Namespace,
+		Name:      infraMachineRef.Name,
+	}
+
+	switch r.Platform {
+	case configv1.AWSPlatformType:
+		infraCluster = &capav1beta2.AWSCluster{}
+		infraMachine = &capav1beta2.AWSMachine{}
+	case configv1.PowerVSPlatformType:
+		infraCluster = &capibmv1.IBMPowerVSCluster{}
+		infraMachine = &capibmv1.IBMPowerVSMachine{}
+	default:
+		return nil, nil, fmt.Errorf("%w: %s", errPlatformNotSupported, r.Platform)
+	}
```

Comment:
This switch will grow quite large, should we factor it out into a separate function?

Reply:
Agreed. Repurposed `getInfraMachineFromProvider` for that too.